src/controller/controller.cpp

```diff
@@ -324,67 +273,56 @@ void Controller::onDialogOK(const EidCertificateAndPinInfo& certAndPinInfo)
         onConfirmCommandHandler(certAndPinInfo);
     } else {
         // This should not happen, and when it does, OK should be equivalent to cancel.
-        onPinPadCancel();
+        onDialogCancel();
     }
 }
 
 void Controller::onDialogCancel()
 {
-    REQUIRE_NON_NULL(window)
-
```

Comment:
Shouldn't you `emit stopCardEventMonitorThread()` here?

Reply:
I do not see that there can be any running thread at this point